### ISSUE TITLE
Allow to create roads

### ISSUE THREAD

**Issue by dev-bre** [TokisanGames/Terrain3D#280]
### Description

I am trying to find a way to add roads (splines) to the terrain, even though I could use other packages, the integration would be limited and the terrain won't change automatically to fill the gap between road and terrain. Is this on the roadmap to add this in this asset directly?

**Comment by TokisanGames** [comment]
We're not providing textures, so there will be no road texture. If the request is to add a spline with non-destructive modification, how is this different from issue #129? https://github.com/TokisanGames/Terrain3D/issues/129#issuecomment-1643492818

**Comment by dev-bre** [comment]
YEah didn't expect to get texture but a way to efficiently cerate roads. So for example, if there is a hill, the tool would allow to create a road which woould follow the shape of the hill and if needed fill the gap between the road segment and the terrain underneat.

**Comment by TokisanGames** [comment]
MTerrain supports non-destructive curves
https://www.youtube.com/watch?v=s1s1zdTGliQ

Currently paths can be made destructively with the height brush or slope brush.

**Comment by JekSun97** [comment]
> Currently paths can be made destructively with the height brush or slope brush.

It's not easy to create a smooth path with brushes without a tablet. Is it possible to consider a function for dofmatija and painting the terrain with splines?

**Comment by TokisanGames** [comment]
> It's not easy to create a smooth path with brushes without a tablet. 

I have no issue making paths or slopes with a mouse, in either 2-click mode or drawable. Only minor smoothing after is required. https://x.com/TokisanGames/status/1750627284204294177/video/1 . Height brush is also useful for this, when using the picker. If you're not using the height picker on the height brush, or the slope brush, and attempting to make paths with raise/lower then yes it would be difficult.

> Is it possible to consider a function for deformation and painting the terrain with splines?

This ticket already exists to consider non-destructive terrain editing with splines. There's no point to destructive with splines. The amount of clicking to get a spline on the ground would be far more than just using the height and slope brushes properly.
